### DISCUSSION TITLE
ClearValues() public method?

### DISCUSSION THREAD

**Discussion by pha3z** [sebastienros/jint#1507] [Ideas]
After calling SetValue() several times to set values and calling a script, I'd like to clear out everything I set, and then set new values before executing another script.  Is there a reason there isn't a ClearValues() method on Engine?

**Comment by lahma** [comment]
Yes,  because there's no tracking what you've set (or replaced). For fresh start either use new engine instance or try ShadowRealm.

**Comment by pha3z** [comment]
How should ShadowRealm be used?

I looked at the Engine constructor, and it looks like there is a *lot* of allocation and setup going when you new one up.   It seems unnecessarily expensive if you just want to reset to a starting state between frequent repeated executions.

Is there are practical way to reset to the starting state?

**Reply by lahma** [reply]
[Here's example of the Shadow Realm](https://github.com/sebastienros/jint/blob/main/Jint.Tests.PublicInterface/ShadowRealmTests.cs).
